src/Raven.Server/Documents/Handlers/Processors/Subscriptions/AbstractSubscriptionsHandlerProcessorForCreateSubscription.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Subscriptions
+{
+    internal abstract class AbstractSubscriptionsHandlerProcessorForCreateSubscription<TRequestHandler, TOperationContext> : AbstractSubscriptionsHandlerProcessorForCreateOrUpdateSubscription<TRequestHandler, TOperationContext>
+        where TOperationContext : JsonOperationContext
+        where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+    {
+        protected AbstractSubscriptionsHandlerProcessorForCreateSubscription([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override bool GetDisabled(SubscriptionCreationOptions options)
+        {
+            return options.Disabled;
+        }
+
+        protected override long? GetSubscriptionId(SubscriptionUpdateOptions options)
+        {
+            return RequestHandler.GetLongQueryString("id", required: false);
+        }
+
+        protected override async ValueTask CreateInternalAsync(SubscriptionUpdateOptions options, TransactionOperationContext context)
+        {
+            var id = GetSubscriptionId(options);
+            await CreateSubscriptionInternalAsync(RequestHandler, id, options, context);
```

Comment:
would it make sense to unify the `CreateSubscriptionInternalAsync` & `CreateInternalAsync` into `CreateSubscriptionAsync` and override it?

Reply:
no you have different implementations for `CreateInternalAsync` in  `AbstractSubscriptionsHandlerProcessorForUpdateSubscription` & `ShardedSubscriptionsHandlerProcessorForCreateSubscription`